src/opm/parser/eclipse/Parser/ParserItem.cpp

```diff
@@ -78,20 +91,22 @@ std::string ParserItem::string_from_size( ParserItem::item_size sz ) {
 }
 
 template<> const int& ParserItem::value_ref< int >() const {
-    if( this->type != get_type< int >() )
-        throw std::invalid_argument( "Wrong type." );
+    if( this->data_type != get_type< int >() )
+        throw std::invalid_argument( "ValueRef<int>: Wrong type." );
     return this->ival;
 }
 
 template<> const double& ParserItem::value_ref< double >() const {
-    if( this->type != get_type< double >() )
-        throw std::invalid_argument( "Wrong type." );
+    if( this->data_type != get_type< double >() ) {
+        printf("type_tag:%d    itype:%d \n", this->data_type, this->input_type);
+        throw std::invalid_argument( " HER .... ValueRef<double> Wrong type." );
```

Comment:
Is this message supposed to say `HER`?  Also, you need `<stdio.h>` for `printf()`.

Reply:
No - removed